lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -0,0 +1,1238 @@
+    """
+    Returns a FHIR Questionnaire Response resource entry for the follow-up
+    encounter.
+
+    Note: `fu_which_activites` is misspelled on
+    purpose to match the misspelling of the fields in the REDCap project.
+    """
+    boolean_questions = [
+        'fu_illness',
+        'fu_change',
+        'result_changes',
+        'fu_household_sick'
+    ]
+
+    integer_questions = [
+        'fu_days_illness',
+        'fu_number_sick'
+    ]
+
+    string_questions = [
+        'fu_fever',
+        'fu_headache',
+        'fu_cough',
+        'fu_chills',
+        'fu_sweat',
+        'fu_throat',
+        'fu_nausea',
+        'fu_nose',
+        'fu_tired',
+        'fu_ache',
+        'fu_breathe',
+        'fu_diarrhea',
+        'fu_rash',
+        'fu_ear',
+        'fu_eye',
+        'fu_smell_taste',
+        'fu_feel_normal',
+        'fu_care',
+        'fu_hospital_ed',
+        'fu_work_school',
+        'fu_activities',
+        'fu_which_activites',
+        'fu_test_result',
+        'fu_behaviors_no',
+        'fu_behaviors_inconclusive',
+        'fu_behaviors',
+        'fu_1_symptoms',
+        'fu_2_symptoms',
+        'fu_3_symptoms',
+        'fu_4_symptoms',
+        'fu_1_test',
+        'fu_2_test',
+        'fu_3_test',
+        'fu_4_test',
+        'fu_1_result',
+        'fu_2_result',
+        'fu_3_result',
+        'fu_4_result',
+        'fu_healthy_test',
+        'fu_healthy_result'
+    ]
+
+    date_questions = [
+        'fu_timestamp',
+        'fu_symptom_duration',
+        'fu_date_care',
+        'fu_1_date',
+        'fu_2_date',
+        'fu_3_date',
+        'fu_4_date',
+        'followup_date'
+    ]
+
+    question_categories = {
+        'valueBoolean': boolean_questions,
+        'valueInteger': integer_questions,
+        'valueString': string_questions,
+        'valueDate': date_questions
+    }
+
+    # Combine checkbox answers into one list
+    checkbox_fields = [
+        'fu_care',
+        'fu_which_activites',
+        'fu_behaviors_no',
+        'fu_behaviors_inconclusive',
+        'fu_behaviors',
+        'fu_1_symptoms',
+        'fu_2_symptoms',
+        'fu_3_symptoms',
+        'fu_4_symptoms'
+    ]
+
+    for field in checkbox_fields:
+        record[field] = combine_checkbox_answers(record, field)
+
+    return questionnaire_response(record, question_categories, patient_reference, encounter_reference)
+
+def create_testing_determination_internal_questionnaire_response(record: dict, patient_reference: dict,
+                                            encounter_reference: dict) -> Optional[dict]:
+    """
+    Returns a FHIR Questionnaire Response resource entry for the
+    testing_determination_internal instrument. This instrument is used
+    to communicate with the Priority Queue application. Note that for this
+    questionnaire we do want to return answers for all questions even if they are
+    not answered.
+    """
```

Comment:
Can you explain why we want to ingest these fields even if they are not answered? Like we discussed earlier today, empty string values violates FHIR guidelines.

Reply:
These fields will be used by the quota/priority queue application. I thought providing them always would make it easier on the quota query side. Unfortunately, the only way to grab data from REDCap is into a FHIR bundle. If we don't want to violate its guidelines and the quote app can do just fine without the fields when they're empty I can remove this functionality.

Reply:
When we don't ingest them because they are empty, they should just show up as null values in the SQL query. I think that matches our expectations.

Reply:
OK. I'll do the surgery to remove the related code. Look for a new commit coming in a bit.

Reply:
I've removed the code that kept items that had an empty string value.

Reply:
Thanks!